packages/lit-html/src/lit-html.ts

```diff
@@ -1080,13 +1080,20 @@ class ChildPart implements Disconnectable {
       ) {
         const parentNodeName = this._$startNode.parentNode?.nodeName;
         if (parentNodeName === 'STYLE' || parentNodeName === 'SCRIPT') {
-          this._insert(
-            new Text(
-              '/* lit-html will not write ' +
-                'TemplateResults to scripts and styles */'
-            )
-          );
-          return;
+          let message = 'Forbidden';
+          if (DEV_MODE) {
+            if (parentNodeName === 'STYLE') {
+              message =
+                `Lit does not support binding inside style nodes. ` +
+                `This is a security risk, as style injection attacks can exfiltrate data and spoof UIs. ` +
+                `Consider instead using css\`...\` literals to compose styles, and accomplishing dynamicism by mutating the DOM rather than styles`;
+            } else {
+              message =
+                `Lit does not support binding inside script nodes. ` +
```

Comment:
This is fine as an existing restriction, but I'd like to dig into this more with the ISE team.

First, in client-side rendering <script> tags in Lit templates will not execute (in SSR they can, but we could make sure they don't).

Then <script> tags can be used for other purposes than executable JS, mainly because they're raw text elements and can contain unescaped "<" characters. The Playground Elements use <script> tags for file content. I've seen syntax highlighters that do the same. I could see cases where you want those to be dynamic...

Reply:
Yeah, we could add some additional checks here, e.g. that there's a `type` field of a non-javascript type.

It is the case that these script tags don't typically run, but are we certain that there isn't a way to make them run? e.g. render into an unattached document fragment, then attach that? Something something inert template and clone into document?

Reply:
I'm not 100% sure if cloning drops the script-created bit, but I'd like to be! Seems like good tests to eventually add.